.eslintrc.js

```diff
@@ -1,22 +1,62 @@
+// @ts-check
+
+/**
+ * @summary The root ESLint configuration.
+ * @description
+ *
+ * - If a workspace emits declaration files, it should contain a `.eslintrc.js`
+ * which extends {@link file://./.config/eslintrc.typed-workspace.js}.
+ * - A workspace _only_ needs its own `.eslintrc.js` if it needs to override this configuration.
+ * - While you may be tempted, please do not use an `eslintConfig` prop in your
+ * `package.json`; use `.eslintrc.js` for consistency.
+ * - Note: glob patterns in this file should _not_ be relative.
+ *
+ * @packageDocumentation
+ */
+
+/**
+ * The minimum EcmaScript runtime environment to support
+ * @see {@link https://eslint.org/docs/latest/use/configure/language-options#specifying-environments}
+ */
+const ECMASCRIPT_ENV = 'es2021'
```

Comment:
There's one area in which we cannot go beyond es2020. I'd have to dig out why, but I remember adding a test to ensure we produce a dist file that's es2020

Now that we moved away from old Node versions this might be irrelevant, but I'm trying to recall how we found out about that issue when we broke it back then.

Reply:
@naugtur The ES2020 check arrived in https://github.com/LavaMoat/LavaMoat/pull/467, but there's little context there as to _why_, though I imagine it's because of whatever browsers MM supports.

Based on the versions mentioned in [this comment](https://github.com/MetaMask/metamask-extension/issues/6805#issuecomment-1398953144) have full support for ES2021, so we should be able to use it if Node.js v16 has full support (which it does).

Thoughts?  I'm not sure where to get more information.

cc @legobeat

Reply:
The ES2020 test was introduced because lavapack runtime is inlined directly into the final bundle, so it dictates what the bundles built with lavamoat support. And MM is supporting (or was at the time) browsers that were missing a feature form 2021. 
If we want to introduce it, it's a big breaking change with a big red warning for users.

This setting is fine for all code other than the runtime files that th test is covering, so it's a matter of skipping linting them which we might be already doing.

Reply:
@legobeat Based on that information, are we OK with this?  the runtime files in lavapack [are _not_ linted](https://github.com/LavaMoat/LavaMoat/blob/9e9a15f29fbdc758fa45e9c8370fa5abe864c7e6/.eslintignore#L8-L10)

Reply:
@naugtur but, would it not make sense to lint those files loosely but restrict the language level to es2020?  I'm guessing that's `lavapack/src/runtime-cjs.js` and `lavapack/src/runtime.js`?

Reply:
since `lavapack/src/builder-runtime.js` asserts `ECMA_VERSION_2020`
before generating `lavapack/src/runtime-cjs.js` and `lavapack/src/runtime.js`
from `lavapack/src/runtime-cjs-template.js` and `lavapack/src/runtime-template.js`

think suggested PR split makes sense? so we bump `lavapack/src/builder-runtime.js` from `ECMA_VERSION_2020` to `ECMA_VERSION_2021` before
- https://github.com/LavaMoat/LavaMoat/pull/1023